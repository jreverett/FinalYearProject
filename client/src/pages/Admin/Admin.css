#admin-container {
  height: 70%;
}

.admin-tab {
  height: 53vh;
}
