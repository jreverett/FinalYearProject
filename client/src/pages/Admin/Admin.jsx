import React, { Component } from 'react';
import { Tabs, Tab } from 'react-bootstrap';
import { UserAdminRow, EventAdminRow } from '../../components';
import { userService, eventService } from '../../services';
import '../../common.css';
import './Admin.css';

class Admin extends Component {
  constructor(props) {
    super(props);

    this.state = {
      tabKey: 'users',
      users: [],
      events: [],
    };
  }

  componentDidMount() {
    this.getUserRows();
    this.getEventRows();
  }

  getUserRows() {
    let userRows = [];
    userService.get().then((users) => {
      users = users.data.sort((a, b) => a.lastname.localeCompare(b.lastname));
      userRows = users.map((user, index) => {
        return (
          <UserAdminRow
            key={index}
            userRowIndex={index}
            user={user}
            updateUserSuspension={this.handleUserSuspensionUpdate}
          />
        );
      });
      this.setState({ users: userRows });
    });
  }

  getEventRows() {
    let eventRows = [];
    eventService.get().then((events) => {
      events = events.data.sort((a, b) => a.title.localeCompare(b.title));
      eventRows = events.map((event, index) => {
        return (
          <EventAdminRow key={index} eventRowIndex={index} event={event} />
        );
      });
      this.setState({ events: eventRows });
    });
  }

  handleUserSuspensionUpdate = (userRowID, suspended) => {
    // update user state
    let newUsers = this.state.users;
    newUsers[userRowID].props.user.suspended = suspended;

    // set new user state
    this.setState({ users: newUsers });
  };

  render() {
    return (
      <div id="admin-container" className="form-container col-md-6 offset-md-3">
        <p className="text-header">Upvent Admin Panel</p>
        <div style={{ height: 'auto' }}>
          <Tabs
            activeKey={this.state.tabKey}
            onSelect={(tabKey) => this.setState({ tabKey })}
          >
            <Tab eventKey="users" title="Users" className="admin-tab">
              {/* user table */}
              <div className="table-container">
                <div>{this.state.users}</div>
              </div>
            </Tab>
            <Tab eventKey="events" title="Events" className="admin-tab">
              {/* event table */}
              <div className="table-container">
                <div>{this.state.events}</div>
              </div>
            </Tab>
          </Tabs>
        </div>
      </div>
    );
  }
}

export default Admin;
