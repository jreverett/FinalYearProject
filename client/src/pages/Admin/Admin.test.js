import React from 'react';
import renderer from 'react-test-renderer';
import Admin from './Admin';

describe('<Admin />', () => {
  it('matches the snapshot', () => {
    const tree = renderer.create(<Admin />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
