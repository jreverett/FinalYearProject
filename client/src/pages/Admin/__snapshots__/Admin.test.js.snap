// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Admin /> matches the snapshot 1`] = `
<div
  className="form-container col-md-6 offset-md-3"
  id="admin-container"
>
  <p
    className="text-header"
  >
    Upvent Admin Panel
  </p>
  <div
    style={
      Object {
        "height": "auto",
      }
    }
  >
    <nav
      className="nav nav-tabs"
      onKeyDown={[Function]}
      role="tablist"
    >
      <a
        aria-controls={null}
        aria-selected={true}
        className="nav-item nav-link active"
        data-rb-event-key="users"
        href="#"
        id={null}
        onClick={[Function]}
        onKeyDown={[Function]}
        role="tab"
      >
        Users
      </a>
      <a
        aria-controls={null}
        aria-selected={false}
        className="nav-item nav-link"
        data-rb-event-key="events"
        href="#"
        id={null}
        onClick={[Function]}
        onKeyDown={[Function]}
        role="tab"
        tabIndex={-1}
      >
        Events
      </a>
    </nav>
    <div
      className="tab-content"
    >
      <div
        aria-hidden={false}
        aria-labelledby={null}
        className="fade admin-tab tab-pane active show"
        id={null}
        role="tabpanel"
      >
        <div
          className="table-container"
        >
          <div />
        </div>
      </div>
      <div
        aria-hidden={true}
        aria-labelledby={null}
        className="fade admin-tab tab-pane"
        id={null}
        role="tabpanel"
      >
        <div
          className="table-container"
        >
          <div />
        </div>
      </div>
    </div>
  </div>
</div>
`;
